transportable-udfs-codegen/src/main/java/com/linkedin/transport/codegen/HiveWrapperGenerator.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.codegen;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import com.linkedin.transport.api.udf.StdUDF;
+import com.linkedin.transport.api.udf.TopLevelStdUDF;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.WildcardTypeName;
+import java.io.File;
+import java.util.Collection;
+import java.util.List;
+import javax.lang.model.element.Modifier;
+
+
+public class HiveWrapperGenerator implements WrapperGenerator {
+
+  private static final String HIVE_PACKAGE_SUFFIX = "hive";
+  private static final String GET_TOP_LEVEL_UDF_CLASS_METHOD = "getTopLevelUdfClass";
+  private static final String GET_STD_UDF_IMPLEMENTATIONS_METHOD = "getStdUdfImplementations";
+  private static final ClassName HIVE_STD_UDF_WRAPPER_CLASS_NAME =
+      ClassName.bestGuess("com.linkedin.transport.hive.StdUdfWrapper");
+
+  @Override
+  public void generateWrappers(ProjectContext context) {
+    Multimap<String, String> udfs = context.getUdfProperties().getUdfs();
+    for (String topLevelStdUDFClass : udfs.keySet()) {
```

Comment:
refactor topLevelStdUDFClass  to topLevelClassName after using getTopLevelClassNames (comment applies to other usages).